test/forked_joins.js

```diff
@@ -0,0 +1,175 @@
+var test = require('tape')
+var hyperlog = require('hyperlog')
+var fdstore = require('fd-chunk-store')
+var path = require('path')
+var memdb = require('memdb')
+var collect = require('collect-stream')
+
+var tmpdir = require('os').tmpdir()
+var storefile = path.join(tmpdir, 'osm-store-' + Math.random())
+
+var osmdb = require('../')
+var versions = {}
+var ways = []
+var osm = osmdb({
+  log: hyperlog(memdb(), { valueEncoding: 'json' }),
+  db: memdb(),
+  store: fdstore(4096, storefile)
+})
+
+test('setup db', function (t) {
+  t.plan(9)
+  var batch = [
+    { type: 'put', key: 'A', value: { type: 'node', lat: 64.5, lon: -147.3 } },
+    { type: 'put', key: 'B', value: { type: 'node', lat: 1.0, lon: 2.0 } },
+    { type: 'put', key: 'C', value: { type: 'node', lat: 3.0, lon: 4.0 } },
+    { type: 'put', key: 'D', value: { type: 'node', lat: 64.123, lon: -147.56 } },
+    { type: 'put', key: 'F', value: { type: 'way', refs: [ 'A', 'B' ] } }
+  ]
+  osm.batch(batch, function (err, nodes) {
+    t.error(err)
+    nodes.forEach(function (node) {
+      versions[node.value.k] = node.key
+    })
+    ways[0] = versions.F
+    console.log('way0 version:', ways[0])
+    ready()
```

Comment:
The indexes ought to be waited on before reading each time, right? So maybe `osm.ready(ready)`?